lib/web_ui/lib/src/engine/semantics/link.dart

```diff
@@ -18,9 +18,8 @@ class Link extends PrimaryRoleManager {
   @override
   DomElement createElement() {
     final DomElement element = domDocument.createElement('a');
-    // TODO(chunhtai): Fill in the real link once the framework sends entire uri.
-    // https://github.com/flutter/flutter/issues/102535.
-    element.setAttribute('href', '#');
+    // TODO: The <a> element has `aria-label={value}`. We should stop that!
+    element.setAttribute('href', semanticsObject.hasValue ? semanticsObject.value! : '#');
```

Comment:
If this is going to use `value` as `href`, then this role manager should not use the `withBasics` super constructor, because that one adds the `LabelAndValue` role which includes `value` in the label. Alternatively, we can make `LabelAndValue` configurable so it does not include `value` in some cases.

Reply:
Fixed it by introducing an allowlist `labelSources` (also used it for incrementable, which was hardcoded as a one-off case).